src/pages/preferences/PreferencesTabs.tsx

```diff
@@ -39,25 +34,56 @@ const Tabs: React.FC<PreferenceTabsProps> = ({
   useEffect(() => {
     dispatchBanner.clearAllBanners();
   }, [tabKey]); // eslint-disable-line react-hooks/exhaustive-deps
-  const Component = componentMap[tabKey];
+  const { title, Component } = getTitleAndComponent(tabKey, userSettings);
   return (
     <Container>
-      <Title data-cy="preferences-tab-title">{mapUrlTabToTitle[tabKey]}</Title>
+      <Title data-cy="preferences-tab-title">{title}</Title>
       <Banners
         banners={bannersState}
         removeBanner={dispatchBanner.removeBanner}
       />
       {loading && <Skeleton active />}
-      {!loading && <Component tabKey={tabKey} {...userSettings} />}
+      {!loading && <Component />}
     </Container>
   );
 };
 
-const componentMap = {
-  profile: ProfileTab,
-  notifications: NotificationsTab,
-  cli: CliTab,
-  newUI: NewUITab,
+const getTitleAndComponent = (
+  tabKey: PreferencesTabRoutes,
```

Comment:
although `tabKey` being an enum provides greater safety, can we still provide it with a default value to be even safer? e.g. `tabKey: PreferencesTabRoutes = PreferencesTabRoutes.default`

Reply:
the default value will only be applied if tabKey argument is undefined. This is pretty good but I see that the enum value mapping in this function could unintentionally could become out of sync with`PreferencesTabRoutes` where `PreferencesTabRoutes` contains more values than the mapping established in this function. To avoid returning `undefined` in this case, I return a default value in the function body